src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIRealtimeExtensions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI.RealtimeConversation;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="RealtimeConversationSession"/> and related types.
+/// </summary>
+public static class OpenAIRealtimeExtensions
+{
+    /// <summary>
+    /// Converts a <see cref="AIFunction"/> into a <see cref="ConversationFunctionTool"/> so that
+    /// it can be used with <see cref="RealtimeConversationClient"/>.
+    /// </summary>
+    /// <returns>A <see cref="ConversationFunctionTool"/> that can be used with <see cref="RealtimeConversationClient"/>.</returns>
+    public static ConversationFunctionTool ToConversationFunctionTool(this AIFunction aiFunction)
+    {
+        _ = Throw.IfNull(aiFunction);
+
+        var parametersSchema = new ConversationFunctionToolParametersSchema
+        {
+            Type = "object",
```

Comment:
Doesn't OpenAI require the type "function" for its top-level schema?

Reply:
This isn't the top-level schema; it's the list of parameters. It's equivalent to this sample code: https://github.com/Azure-Samples/aoai-realtime-audio-sdk/blob/main/dotnet/samples/console-from-file/Program.cs#L230-L249

So, `object` is apparently the right value here and does work.